File: dotnet/xmsclient/Client/Protocol/AMQMethodEvent.cs
using System;
using System.Text;
using jpmorgan.mina.common;
using log4net;
using OpenAMQ.Framing;

namespace OpenAMQ.XMS.Client.Protocol
{
    public class AMQMethodEvent
    {
        private AMQMethodBody _method;

        private ushort _channelId;

        private AMQProtocolSession _protocolSession;

        public AMQMethodEvent(ushort channelId, AMQMethodBody method, AMQProtocolSession protocolSession)
        {
            _channelId = channelId;
            _method = method;
            _protocolSession = protocolSession;
        }

        public AMQMethodBody Method
        {
            get
            {
                return _method;
            }
        }

        public ushort ChannelId
        {
            get
            {
                return _channelId;
            }
        }

        public AMQProtocolSession ProtocolSession
        {
            get
            {
                return _protocolSession;
            }            
        }

        public override String ToString()
        {
            StringBuilder buf = new StringBuilder("Method event: ");
            buf.Append("\nChannel id: ").Append(_channelId);
            buf.Append("\nMethod: ").Append(_method);
            return buf.ToString();
        }
    }
}

File: java/client/src/org/openamq/client/protocol/AMQMethodEvent.java
package org.openamq.client.protocol;

import org.openamq.framing.AMQMethodBody;

/**
 * @author Robert Greig (robert.j.greig@jpmorgan.com)
 */
public class AMQMethodEvent
{
    private AMQMethodBody _method;

    private int _channelId;
    
    private AMQProtocolSession _protocolSession;

    public AMQMethodEvent(int channelId, AMQMethodBody method, AMQProtocolSession protocolSession)
    {
        _channelId = channelId;
        _method = method;
        _protocolSession = protocolSession;
    }

    public AMQMethodBody getMethod()
    {
        return _method;
    }

    public int getChannelId()
    {
        return _channelId;
    }
    
    public AMQProtocolSession getProtocolSession()
    {
        return _protocolSession;
    }
    
    public String toString()
    {
        StringBuffer buf = new StringBuffer("Method event: ");
        buf.append("\nChannel id: ").append(_channelId);
        buf.append("\nMethod: ").append(_method);
        return buf.toString();
    }
}
